services/worker/src/worker/job_runners/parquet_and_dataset_info.py

```diff
@@ -528,10 +528,28 @@ def raise_if_too_big_from_external_data_files(
         base_path=builder.base_path, download_config=DownloadConfig(use_auth_token=hf_token)
     )
     try:
-        builder._split_generators(mock_dl_manager)
-    except NotImplementedError as err:
-        if "is not implemented in streaming mode." not in str(err):
-            raise
+        builder._split_generators(mock_dl_manager)  # type: ignore
+    except (requests.exceptions.RequestException, NotImplementedError) as error:
+        if isinstance(error, NotImplementedError):
+            # we can ignore the errors from functions not implemented in streaming mode like `.extract()` on TAR files
+            if "is not implemented in streaming mode." not in str(error):
+                raise
```

Comment:
as we raise, it will be caught and transformed into an "Unexpected error" when shown to the user in the dataset viewer with the raw message. That's why we always try to catch the exceptions and create a specific message that will help the user understand what they can do on their side to fix the issue or, else, how they can report it.

cc @albertvillanova

Reply:
Ok, good to know. Should I add an dedicated error here as well then ?

Reply:
Yes, I think so

Reply:
We really should have Unexpected errors only for the cases we don't expect and that indicate a bug in the datasets-server code.

Reply:
I agree, for the end user is better to have actionable error messages.